deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: gitsources.devopsconsole.openshift.io
+spec:
+  group: devopsconsole.openshift.io
+  names:
+    kind: GitSource
+    listKind: GitSourceList
+    plural: gitsources
+    singular: gitsource
+    shortNames:
+      - gitsrc
+  scope: Namespaced
+  validation:
+    # openAPIV3Schema is the schema for validating custom objects.
+    openAPIV3Schema:
+      properties:
+        apiVersion:
+          type: string
+        kind:
+          type: string
+        metadata:
+          type: object
+        spec:
+          properties:
+            url:
+              type: string
```

Comment:
Today I was talking to @tinakurian who is writing the build tool detector. While talking we realized that the Import and Create flow are literally the same thing in the beginning of the flow and end potentially differently. Both flows clone a repo from github in the beginning. In the end, they push either a devfile (import flow) or a complete code package to a git remote.

Now, because of this I suggest to reflect the similarity between these two here.

I cannot recall directly but do you want to use two GitSource CRs for cloning and pushing? If not and you only want to use one, I suggest to have two URLs here, one for pull and one for push. The reason is that for an import flow, we should already know what credentials and transport protocol we need in order to clone a repo. Remember, that during an import, we just clone a github booster repo which will eventually become the new project that we push somewhere else.

Make sense?

@sbose78 @michaelkleinhenz thoughts?

Reply:
Replayed here - https://jira.coreos.com/browse/ODC-65?focusedCommentId=70213&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-70213
TL;DR I think we should use multiple GitSource CRs if/when needed.

Reply:
oh, okay then.